sdk/python/jobs/finetuning/standalone/chat-completion/chat_completion_with_model_as_service.ipynb

```diff
@@ -0,0 +1,559 @@
+    "\n",
+    "**Note** : Some language models have different language codes and hence the column names in the dataset should reflect the same.\n",
+    "\n",
+    "The chat-completion dataset is stored in parquet format with each entry using the following schema:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "    {\n",
+    "        \"prompt\": \"Create a fully-developed protagonist who is challenged to survive within a dystopian society under the rule of a tyrant. ...\",\n",
+    "        \"messages\":[\",\n",
+    "            {\",\n",
+    "                \"content\": \"Create a fully-developed protagonist who is challenged to survive within a dystopian society under the rule of a tyrant. ...\",\n",
+    "                \"role\": \"user\",\n",
+    "            },\n",
+    "            {\",\n",
+    "                \"content\": \"Name: Ava\\n Ava was just 16 years old when the world as she knew it came crashing down. The government had collapsed, leaving behind a chaotic and lawless society. ...\",\n",
+    "                \"role\": \"assistant\",\n",
+    "            },\n",
+    "            {\",\n",
+    "                \"content\": \"Wow, Ava's story is so intense and inspiring! Can you provide me with more details.  ...\",\n",
+    "                \"role\": \"user\",\n",
+    "            },\n",
+    "            {\n",
+    "                \"content\": \"Certainly! ....\",\n",
+    "                \"role\": \"assistant\"\",\n",
+    "            }\n",
+    "        ],\n",
+    "        \"prompt_id\": \"d938b65dfe31f05f80eb8572964c6673eddbd68eff3db6bd234d7f1e3b86c2af\",\n",
+    "    }"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install dependencies for downloading datasets from huggingface\n",
+    "\n",
+    "%pip install datasets --upgrade\n",
+    "%pip install py7zr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# download the dataset using the helper script. This needs datasets library: https://pypi.org/project/datasets/\n",
+    "# For demo purposes, we are downloading only 1% of the dataset.\n",
+    "import os\n",
+    "import shutil\n",
+    "\n",
+    "dataset_dir = \"ultrachat_200k_dataset\"\n",
+    "shutil.rmtree(dataset_dir, ignore_errors=True)\n",
+    "exit_status = os.system(\n",
+    "    f\"python ./download-dataset.py --dataset HuggingFaceH4/ultrachat_200k --download_dir {dataset_dir} --dataset_split_pc 1\"\n",
+    ")\n",
+    "if exit_status != 0:\n",
+    "    raise Exception(\"Error downloading dataset\")\n",
+    "\n",
+    "os.rename(f\"./{dataset_dir}/train_sft.jsonl\", f\"./{dataset_dir}/train.jsonl\")\n",
+    "os.rename(f\"./{dataset_dir}/test_sft.jsonl\", f\"./{dataset_dir}/validation.jsonl\")"
```

Comment:
it might be good to add a reason for having validation data produced, in the Prepare data section